File: blog-api/src/main/scala/su/wps/blog/models/domain/Testimonial.scala
package su.wps.blog.models.domain

import io.circe.Encoder

import java.time.ZonedDateTime

final case class Testimonial(
  authorName: String,
  authorTitle: Option[String],
  authorCompany: Option[String],
  authorUrl: Option[String],
  authorImageUrl: Option[String],
  quote: String,
  sortOrder: Int,
  isActive: Boolean,
  createdAt: ZonedDateTime,
  id: Option[TestimonialId] = None
) {
  def nonEmptyId: TestimonialId =
    id.getOrElse(throw new IllegalStateException("Empty testimonial id"))
}

final case class TestimonialId(value: Int) extends AnyVal

object TestimonialId {
  implicit val encoder: Encoder[TestimonialId] = Encoder[Int].contramap(_.value)
}

File: blog-admin/blog_admin/blog_admin/models/testimonial.py
from django.db import models


class Testimonial(models.Model):
    author_name = models.CharField(max_length=255)
    author_title = models.CharField(max_length=255, blank=True, null=True)
    author_company = models.CharField(max_length=255, blank=True, null=True)
    author_url = models.CharField(max_length=500, blank=True, null=True)
    author_image_url = models.CharField(max_length=500, blank=True, null=True)
    quote = models.TextField()
    sort_order = models.IntegerField(default=0)
    is_active = models.BooleanField(default=True)
    created_at = models.DateTimeField(auto_now_add=True)

    class Meta:
        managed = False
        db_table = 'testimonials'
        ordering = ['sort_order']

    def __str__(self):
        company_info = f" ({self.author_company})" if self.author_company else ""
        return f"{self.author_name}{company_info}"
